source/common/upstream/upstream_impl.cc

```diff
@@ -222,6 +222,17 @@ void HostSetImpl::updateHosts(UpdateHostsParams&& update_hosts_params,
   healthy_hosts_per_locality_ = std::move(update_hosts_params.healthy_hosts_per_locality);
   degraded_hosts_per_locality_ = std::move(update_hosts_params.degraded_hosts_per_locality);
   locality_weights_ = std::move(locality_weights);
+
+  rebuildLocalityScheduler(locality_scheduler_, locality_entries_, *healthy_hosts_per_locality_,
+                           *healthy_hosts_);
+
+  runUpdateCallbacks(hosts_added, hosts_removed);
+}
+
+void HostSetImpl::rebuildLocalityScheduler(
+    std::unique_ptr<EdfScheduler<LocalityEntry>>& locality_scheduler,
+    std::vector<std::shared_ptr<LocalityEntry>>& locality_entries,
+    const HostsPerLocality& eligible_hosts_per_locality, const HostVector& hosts) {
```

Comment:
Here or in the follow-up, can you comment member variables for this function as you did above?  It might be worth renaming hosts to something which makes the relationship between eligible_hosts_per_locality and hosts more clear.

I did spend a few minutes making sure we were doing the right thing w.r.t. member variables passed in vs member variables used directly.  It might be more clear if we made this const or even static (passing in hosts_per_locality_ (as all_hosts_per_locality?) and locality_weights_)

Reply:
I'll try making this completely static to see how it looks - I actually spent 30 minutes debugging an issue while writing the other PR that was caused by my leaving the `_` at the end of `locality_scheduler`, so I definitely understand the concern.